addons/mail/models/mail_message.py

```diff
@@ -855,6 +857,19 @@ def _message_format(self, fnames, format_reply=True, legacy=False):
                 'id': message_sudo.author_guest_id.id,
                 'name': message_sudo.author_guest_id.name,
             } if message_sudo.author_guest_id else [('clear',)]
+            # sender: used to display impersonation, mainly for internal users
+            # e.g. when a template is used on behalf of someone else; mail gateway
+            # messages currently consider the author being the sender (as otherwise
+            # it will always be 'Sent by Odoobot')
+            if message_sudo.message_type == 'email':
+                sender = author
```

Comment:
For this part also I would remove the condition here and move it to `senderView` `compute` in JS, so that returned `sender` is always consistently `create_uid`'s partner.